source/validate.cpp

```diff
@@ -332,14 +270,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
+  if (inst.result_id) {
+    _.usedefs().AddDef(
+        {inst.result_id, inst.opcode,
+         std::vector<uint32_t>(inst.words, inst.words + inst.num_words)});
+  }
+  for (auto op = inst.operands; op != inst.operands + inst.num_operands; ++op) {
+    if (op->type == SPV_OPERAND_TYPE_ID)
```

Comment:
Also, I'd argue this capturing of entry points is logically separate from collecting uses and defs.
It's arguable whether it even merits its own function, since the right solution is a single line of code.